### PR TITLE
Feature/time rotation

### PR DESCRIPTION
**Pull request by vche**
Implements #34

*Description of changes:*
Implements time based file rotation 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: flowgger.toml [hunk 1]
```diff
@@ -85,9 +85,19 @@ type = "file"
...
+
 # Optional: Enables file rotation once the specified size is reached.
-file_rotation_size = 980
-# Optional, only used if file_rotation_size is set: Specifies number of rotation files to use. The default value is 50.
+file_rotation_size = 2048
+
+# Optional: Enables file rotation based on time. Rotation occur every file_rotation_time minutes
+file_rotation_time = 2
+
+# Optional: When time rotation is enabled, format of the timestamp added to the filenames.
```
Comment (crisidev) [inline flowgger.toml]: Not very clear message. I would rephrase with

`Optional: When time rotation is enabled, the timestamp format is appended to the filenames.`

File: src/flowgger/utils/rotating_file.rs [hunk 2]
```diff
@@ -143,6 +230,97 @@ impl RotatingFile {
...
+    /// Execute a log file rotation for time triggers
+    /// Create a new file with a timestamped filename. The filename therefore indicates the creation time
+    /// The previous file(s) being also timestamped, they are close
+    ///
+    /// # Returns
+    /// - 'Ok':   when the rotation has been done
+    /// - 'Err':  when the new file could not be open
+    ///
+    fn rotate_time(&mut self) -> io::Result<()> {
+        let _ = writeln!(
```
Comment (crisidev) [inline src/flowgger/utils/rotating_file.rs:246]: Do you really need this `let _` ?

**Review by crisidev — APPROVED** [review]
Two small comments, LGTM!

File: src/flowgger/output/file_output.rs [hunk 3]
```diff
@@ -61,6 +71,16 @@ impl FileOutput {
                     as usize
             },
         );
+        // Get the optional file rotation ti;e. if none, set it to 0 to disable the feature
```
Comment (kurojishi) [inline src/flowgger/output/file_output.rs]: typo: s/ti;e/time/

File: src/flowgger/utils/rotating_file.rs (hunk 2 above)
Comment (vche) [reply src/flowgger/utils/rotating_file.rs:246]: yes, there's warning otherwise because i don't use the writeln status

File: flowgger.toml (hunk 1 above)
Comment (vche) [reply flowgger.toml]: alright

File: src/flowgger/utils/rotating_file.rs [hunk 4]
```diff
@@ -29,42 +62,96 @@ impl RotatingFile {
...
+        new_file.set_file_name(&format!("{}-{}.{}",
+                                self.basename.file_stem().unwrap_or_else(|| OsStr::new("")).to_string_lossy(),
+                                dt_str,
+                                self.basename.extension().unwrap_or_else(|| OsStr::new("")).to_string_lossy()));
+        new_file
+    }
+
     /// Open the base file and ready for logging and set it as current file
     /// Should typically be called after object creation in order to be able to start writing
+    /// The file opened is the basename specified at object creation.
```
Comment (kurojishi) [inline src/flowgger/utils/rotating_file.rs]: This section is not too clear can you reword it a bit?
Something like:

```
It creates a file, if it doesn't exist, and open for writing at the path pointed to in the basename field of the object
```
I'm struggling to put it into clearer wording myself really

File: src/flowgger/utils/rotating_file.rs [hunk 5]
```diff
@@ -182,6 +358,13 @@ mod tests {
     use super::*;
     extern crate tempdir;
     use tempdir::TempDir;
+    use crate::flowgger::utils::test_utils::rfc_test_utils::utc_from_date_time;
+
+//    pub fn get_test_date_time() -> DateTime<Utc> {
+//        println!("time test ");
+//        utc_from_date_time(2015, 8, 6, 11, 15, 24, 637)
+////        fake_time
+//    }
```
Comment (kurojishi) [inline src/flowgger/utils/rotating_file.rs]: comment left from the development?

File: src/flowgger/utils/rotating_file.rs (hunk 4 above)
Comment (vche) [reply src/flowgger/utils/rotating_file.rs]: reworded, added examples

File: src/flowgger/utils/rotating_file.rs (hunk 5 above)
Comment (vche) [reply src/flowgger/utils/rotating_file.rs]: oups....

**Review by kurojishi — APPROVED** [review]
